pkg/config/types/v1alpha1/v1alpha1_configurator.go

```diff
@@ -108,14 +108,12 @@ func (m *MachineConfig) Files() []machine.File {
 
 // Type implements the Configurator interface.
 func (m *MachineConfig) Type() machine.Type {
-	switch m.MachineType {
-	case "init":
-		return machine.TypeInit
-	case "controlplane":
-		return machine.TypeControlPlane
-	default:
+	t, err := machine.ParseType(m.MachineType)
```

Comment:
I don't know of a good way to prevent having to return an error here. If we change the signature, things get super ugly in places we need to know the machine type. I was thinking that we could do `machine.ParseType` in the validation function, that way we know that we will not get an error here, and we can ignore it.